pkg/services/ngalert/notifier/channels/telegram.go

```diff
@@ -0,0 +1,156 @@
+package channels
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"mime/multipart"
+	"net/url"
+
+	gokit_log "github.com/go-kit/kit/log"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	old_notifiers "github.com/grafana/grafana/pkg/services/alerting/notifiers"
+)
+
+const (
+	telegramAPIURL = "https://api.telegram.org/bot%s/sendMessage"
+)
+
+// TelegramNotifier is responsible for sending
+// alert notifications to Telegram.
+type TelegramNotifier struct {
+	old_notifiers.NotifierBase
+	BotToken    string
+	ChatID      string
+	Message     string
+	log         log.Logger
+	tmpl        *template.Template
+	externalUrl *url.URL
+}
+
+// NewTelegramNotifier is the constructor for the Telegram notifier
+func NewTelegramNotifier(model *models.AlertNotification, t *template.Template, externalUrl *url.URL) (*TelegramNotifier, error) {
+	if model.Settings == nil {
```

Comment:
This is the first time I see this validation, should we also add it to the others?

Reply:
I am basically copy pasting existing code and making changes. So looks like we will need this for others too which is missing currently. I will add in the other PRs and update this PR for the other notifiers.